app/src/userpages/components/Header/AccountsBalance/index.jsx

```diff
@@ -0,0 +1,44 @@
+// @flow
+
+import React, { useMemo } from 'react'
+import BN from 'bignumber.js'
+
+import Balance from '$userpages/components/Balance'
+import { BalanceType } from '$shared/flowtype/integration-key-types'
+import { useBalances } from '$shared/hooks/useBalances'
+
+import styles from './accountsBalance.pcss'
+
+const AccountsBalance = () => {
+    const { balances } = useBalances()
+
+    const { ethBalance, dataBalance } = useMemo(() => {
+        const { eth, data } = Object.values(balances).reduce(({ eth: prevEth, data: prevData }, innerBalance: Object) => ({
+            eth: prevEth.plus(innerBalance[BalanceType.ETH] || '0'),
```

Comment:
Could you add a comment here explaining how the balance is updated?

Reply:
Yep, added some more comments in https://github.com/streamr-dev/streamr-platform/pull/896/commits/12ffee77027f67ca801afe85db88022ae77ac9f2